### PR TITLE
feat: Support unicode strings in search

### PR DESCRIPTION
**Pull request by dranikpg**
This PR adds support for unicode strings in search. Those are now split by word boundaries with the ICU library, so unicode sentences should be handled correctly. Added tests  with simple sentences in four languages

Adds a dependency on libicu-dev

### CODE REVIEW

**Review by royjacobson — APPROVED** [review]

File: src/core/search/indices.cc [hunk 1]
```diff
@@ -20,20 +23,64 @@ using namespace std;
...
-  for (auto it = begin; it != end; ++it) {
-    auto word = it->str();
-    absl::AsciiStrToLower(&word);
-    words.insert(move(word));
+// Get all words from text as matched by the ICU library
+absl::flat_hash_set<std::string> ICUTokenizeWords(std::string_view text) {
+  // Is text contains only ascii, skip working with ICU resources
+  if (IsAllAscii(text)) {
+    std::regex rx{"\\b.*?\\b", std::regex_constants::icase};
+    std::cregex_iterator begin{text.data(), text.data() + text.size(), rx}, end{};
```
Comment (kostasrim) [inline src/core/search/indices.cc:35]: why not `text.end()`

File: src/core/search/indices.cc [hunk 2]
```diff
@@ -20,20 +23,64 @@ using namespace std;
...
+
+    absl::flat_hash_set<string> words;
+    for (auto it = begin; it != end; ++it) {
```
Comment (kostasrim) [inline src/core/search/indices.cc:38]: Why not rage-for loop here? 

P.s. you are now on the other side of the mafia  `++it` 🤣

File: src/server/search/search_family_test.cc [hunk 3]
```diff
@@ -287,4 +287,23 @@ TEST_F(SearchFamilyTest, SimpleUpdates) {
...
 }
 
+TEST_F(SearchFamilyTest, Unicode) {
+  EXPECT_EQ(Run({"ft.create", "i1", "schema", "title", "text", "visits", "numeric"}), "OK");
+
+  // Explicitly using screaming uppercase to check utf-8 to lowercase functionality
+  Run({"hset", "d:1", "title", "Веселая СТРЕКОЗА Иван", "visits", "400"});
+  Run({"hset", "d:2", "title", "Die fröhliche Libelle Günther", "visits", "300"});
+  Run({"hset", "d:3", "title", "השפירית המהירה יעקב", "visits", "200"});
+  Run({"hset", "d:4", "title", "πανίσχυρη ΛΙΒΕΛΛΟΎΛΗ Δίας", "visits", "100"});
```
Comment (kostasrim) [inline src/server/search/search_family_test.cc:297]: Greek! Means strong dragonfly zeus 🤣 did you google translate this? (loved ALL the easter egg 👀 )

File: src/server/search/search_family_test.cc [hunk 4]
```diff
@@ -287,4 +287,23 @@ TEST_F(SearchFamilyTest, SimpleUpdates) {
   }
 }
 
+TEST_F(SearchFamilyTest, Unicode) {
+  EXPECT_EQ(Run({"ft.create", "i1", "schema", "title", "text", "visits", "numeric"}), "OK");
+
+  // Explicitly using screaming uppercase to check utf-8 to lowercase functionality
+  Run({"hset", "d:1", "title", "Веселая СТРЕКОЗА Иван", "visits", "400"});
```
Comment (kostasrim) [inline src/server/search/search_family_test.cc:294]: "Merry DRAGONFLY Ivan" 🤣

**Review by kostasrim — APPROVED** [review]
looks good 👨‍🍳

File: src/core/search/indices.cc (hunk 1 above)
Comment (dranikpg) [reply src/core/search/indices.cc:35]: I need a pointer, not an iterator

File: src/core/search/indices.cc (hunk 2 above)
Comment (dranikpg) [reply src/core/search/indices.cc:38]: 1. you can't easily do a for loop if you have two separate iterator -> the for loop construct calls itself begin() and end() functions - on what object in this case?
2. It makes no difference for integers, for iterators I usually pre-increment (not sure if even generic iterators need to have post increment)

File: src/server/search/search_family_test.cc (hunk 3 above)
Comment (dranikpg) [reply src/server/search/search_family_test.cc:297]: Yes, I was just mixing adjective + dragonfly + some stereotypical name 🙂 Sounds like children stories

**Review by kostasrim — APPROVED** [review]

**Pull request merged**